apps/evm/src/clients/api/queries/getPendingRewards/useGetPendingRewards.ts

```diff
@@ -74,14 +76,62 @@ const useGetPendingRewards = (input: TrimmedGetPendingRewardsInput, options?: Pa
     accountAddress: input.accountAddress || undefined,
   });
 
-  const isolatedPoolComptrollerAddresses = useMemo(
-    () =>
-      (getPoolsData?.pools || []).reduce<string[]>(
-        (acc, pool) => (pool.isIsolated ? [...acc, pool.comptrollerAddress] : acc),
-        [],
-      ),
-    [getPoolsData?.pools],
-  );
+  const { isolatedPoolComptrollerAddresses, merklCampaignsPerPool } = useMemo(() => {
+    const data = (getPoolsData?.pools || []).reduce<{
+      isolatedPoolComptrollerAddresses: string[];
+      merklCampaignsPerPool: GetPendingRewardsInput['merklCampaignsPerPool'];
+    }>(
+      (acc, pool) => {
+        const comptrollerAddresses = pool.isIsolated
+          ? [...acc.isolatedPoolComptrollerAddresses, pool.comptrollerAddress]
+          : acc.isolatedPoolComptrollerAddresses;
+
+        // list all Merkl rewards for each market present in the pool
+        const { assets } = pool;
+        const merklRewardsPerAsset = assets.reduce<Record<string, MerklDistribution[]>>(
+          (accMerklDistributionsForAsset, a) => {
+            const assetMerklRewards = [
+              ...a.supplyTokenDistributions.filter(d => d.type === 'merkl'),
+              ...a.borrowTokenDistributions.filter(d => d.type === 'merkl'),
+            ];
+
+            const entries =
+              assetMerklRewards.length > 0
+                ? {
+                    ...accMerklDistributionsForAsset,
+                    [a.vToken.address]: assetMerklRewards,
+                  }
+                : accMerklDistributionsForAsset;
+
+            return entries;
+          },
+          {},
+        );
+
+        // and then classify them under the pool (if there are any)
+        const merklRewards =
+          Object.keys(merklRewardsPerAsset).length > 0
+            ? {
+                ...acc.merklCampaignsPerPool,
+                [pool.comptrollerAddress]: {
+                  ...merklRewardsPerAsset,
+                },
+              }
```

Comment:
I think you can remove the top layer of that object, which lists all the assets with Merkl rewards for each pool using the comptroller address as key, since it isn't used afterwards (other pieces of logic only use the vToken addresses it seems).